### PR TITLE
App 20189 add step to integrate api docs to algolia index

### PR DESCRIPTION
**Pull request by webspaceadam**
## Summary

- postbuild script refactored
- created algoliarecord generators to have an implementable interface for the future and have a maintainable design
- implemented vitest and tests for the generators to not push garbage to algolia

### CODE REVIEW

File: packages/swirl-tokens/dart/lib/styles.dark.dart [hunk 1]
```diff
@@ -4,7 +4,7 @@
 //
 
 // Do not edit directly
-// Generated on Fri, 15 Dec 2023 15:44:18 GMT
```
Comment (fellmann) [inline packages/swirl-tokens/dart/lib/styles.dark.dart:7]: **question:** this is a bit annoying ... can the timestamp be removed somehow?

File: packages/swirl-icons/dart/lib/fonts/swirl-icons.ttf
Comment (fellmann) [inline packages/swirl-icons/dart/lib/fonts/swirl-icons.ttf:1]: **question:** this is a bit annoying ... why do these file change?

File: apps/swirl-docs/tests/scripts/AlgoliaDataGenerator/AlgoliaDataFactory.test.ts
Comment (fellmann) [inline apps/swirl-docs/tests/scripts/AlgoliaDataGenerator/AlgoliaDataFactory.test.ts:1]: **question:** do we want to keep tests in a separate directory, or next to the source files?

**Review by fellmann — APPROVED** [review]
lgtm

File: packages/swirl-tokens/dart/lib/styles.dark.dart (hunk 1 above)
Comment (webspaceadam) [reply packages/swirl-tokens/dart/lib/styles.dark.dart:7]: my current state is that this is not possible unfortunately. but the answer for why maybe has @Sqrrl

File: packages/swirl-icons/dart/lib/fonts/swirl-icons.ttf
Comment (webspaceadam) [reply packages/swirl-icons/dart/lib/fonts/swirl-icons.ttf:1]: part of the build / dev step

File: apps/swirl-docs/tests/scripts/AlgoliaDataGenerator/AlgoliaDataFactory.test.ts
Comment (webspaceadam) [reply apps/swirl-docs/tests/scripts/AlgoliaDataGenerator/AlgoliaDataFactory.test.ts:1]: Yeah i thought about that. But as the whole app will grow i thought having them in a seperate directory keeps the source code "clean". but no hard feelings here. we can also change that :) what is your take on that?

**Pull request merged**